ffi/util.lua

```diff
@@ -313,7 +318,9 @@ function util.idiv(a, b)
     return (q > 0) and math.floor(q) or math.ceil(q)
 end
 
-function util.orderedPairs(t)
+--- Equivalent of the pairs() function on tables. Allows to iterate in order.
+function util.orderedPairs(t) return end
+do
```

Comment:
What/why is that `do` doing there? And why an empty function ? (must be explained in comments if there's an explanation, cause I would wonder :)

Reply:
It's not really something that needs explanation. Sticking something in `do…end` limits the scope. ;-)

The luacheck complaint is valid in my eyes:
```
    ffi/util.lua:317:39: shadowing upvalue argument 't' on line 316
    ffi/util.lua:327:32: shadowing upvalue argument 't' on line 316
```
That means that while reading the code, you have to be really careful or you might get confused about the specifics of what `t` is. I don't like that, and I suspect neither does anybody else. So by using `do…end` I'm doing the same thing as embedding it in the function. Nothing outside that scope has access to it.

Normally I'd probably just write `local _whatever` (with `_` clearly indicating local use only elsewhere in the code) but in this case it seemed fairly elegant. Anyway, happy to change it. It's just one of those "all ways lead to Rome" kind of things. :-P

Reply:
Thanks for the explanation, didn't know that trick. May be just add:
`do -- just to limit the scope of the following, and not pollute namespace` or something :)